php/class-post-type.php

```diff
@@ -549,4 +552,86 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
+		if ( did_action( 'customize_save' ) ) {
+			// Short circuit because customize_save ajax call is changing status.
+			return;
+		}
+		$snapshot_theme = get_post_meta( get_the_ID(), '_snapshot_theme', true );
+		if ( ! empty( $snapshot_theme ) && get_stylesheet() !== $snapshot_theme ) {
```

Comment:
@valendesigns What were the reasons for preventing snapshots from being published when doing a theme switch? I can't recall. Is it because themes may define their own settings, and if a different theme is active when a snapshot is loaded or saved, it could result in there being unrecognized settings?